style: apply /style-guide pass to models/reports (#2687)
wandb/docs: models/reports/edit-a-report.mdx

description: Edit a report interactively with the App UI or programmatically with
  the W&B SDK.
title: Edit a report
keywords: ["wandb-workspaces", "PanelGrid", "RunSet", "report blocks", "report filters"]
---

<Note>
W&B Report and Workspace API is in Public Preview.
</Note>

This page describes how to edit a report interactively with the App UI or programmatically with the W&B SDK. It covers adding plots, run sets, code blocks, markdown, HTML elements, and rich media, as well as filtering and grouping run sets, organizing report layout, and visualizing multi-dimensional relationships.

> Reports consist of _blocks_. Blocks make up the body of a report. Within these blocks you can add text, images, embedded visualizations, plots from experiments and runs, and panel grids.

Reviewer: ```suggestion
A report's body consists of _blocks_. Blocks contain text, images, embedded visualizations, plots from experiments and runs, and panel grids.
```
Aren't all of the things from "embedded" to "runs" panel grids? Seems redundant at the end.

_Panel grids_ are a specific type of block that hold panels and _run sets_. Run sets are a collection of runs logged to a project in W&B. Panels are visualizations of run set data.

<Note>

> See the [Programmatic workspaces notebook](https://colab.research.google.com/github/wandb/wandb-workspaces/blob/Update-wandb-workspaces-tuturial/Workspace_tutorial.ipynb) for a step-by-step example of how to create and customize a saved workspace view.

Reviewer: ```suggestion
Run the [Programmatic workspaces notebook](https://colab.research.google.com/github/wandb/wandb-workspaces/blob/Update-wandb-workspaces-tuturial/Workspace_tutorial.ipynb) for an end-to-end tutorial of creating a customized saved workspace view.
```

runset = wr.Runset(
  project=project,
  entity=entity,
  groupby=["summary.acc"]  # Group by summary values
)
```

## Filter a run set programmatically

Programmatically filter run sets and add them to a report with the [Workspace and Reports API](/models/ref/wandb_workspaces/reports). Filtering narrows a run set to the specific runs you want to display, based on config values, metrics, tags, or run properties.

The general syntax for a filter expression is:

```text
Filter('KEY') operation [VALUE]
```

In this expression, `KEY` is the name of the filter, `operation` is a comparison operator (for example, `>`, `<`, `==`, `in`, `not in`, `or`, and `and`), and `[VALUE]` is the value to compare against. `Filter` is a placeholder for the type of filter you want to apply. The following table lists the available filters and their descriptions:

| Filter | Description | Available keys |
| ---|---| --- |

> |`Config('KEY')` | Filter by config values | Values specified in `config` parameter in `wandb.init(config=)`. |

Reviewer: Should these be `'[KEY]'` or is it too confusing?

<Tabs>
<Tab title="App UI">
Enter a forward slash (`/`) in the report to display a dropdown menu. From the dropdown, choose **Code**.

Select the name of the programming language on the right side of the code block to expand a dropdown. From the dropdown, select your programming language syntax. You can choose from JavaScript, Python, CSS, JSON, HTML, Markdown, and YAML.
</Tab>
<Tab title="Report and Workspace API">
Use the `wr.CodeBlock` Class to create a code block programmatically. Provide the name of the language and the code you want to display for the language and code parameters, respectively.

> For example, the following example demonstrates a list in a YAML file:

Reviewer: ```suggestion
The following example demonstrates a list in a YAML file:
```

## Duplicate and delete panel grids

> Duplicating and deleting panel grids helps you reuse layouts and remove content you no longer need. If you have a layout that you want to reuse, select a panel grid and copy-paste it to duplicate it in the same report or paste it into a different report.

Reviewer: This section should be split up. It's odd to combine duplicating and deleting.

    <img src="/images/reports/delete_panel_grid.gif" alt="Deleting panel grids"  />
</Frame>

## Collapse headers to organize reports

Collapse headers in a report to hide content within a text block. When the report loads, only expanded headers show content. Collapsing headers in reports helps organize your content and prevents excessive data loading. The following gif demonstrates the process.

<Frame>
    <img src="/images/reports/collapse_headers.gif" alt="Collapsing headers in a report."  />
</Frame>

## Visualize relationships across multiple dimensions

> When you want to compare more variables than a 2D plot can show, you can use a color gradient as an additional dimension. W&B recommends using a color gradient to represent one of the variables, which enhances clarity and makes patterns easier to interpret.

Reviewer: ```suggestion
To compare more variables than a 2D plot can show, you can use a color gradient as an additional dimension. Using a color gradient to represent one of the variables can make patterns easier to interpret.
```

> 1. Choose a variable to represent with a color gradient (for example, penalty scores or learning rates). This provides a clearer understanding of how penalty (color) interacts with reward/side effects (y-axis) over training time (x-axis).

Reviewer: ```suggestion
1. Choose a variable to represent with a color gradient, like penalty scores or learning rates. This provides a clearer understanding of how penalty (color) interacts with reward/side effects (y-axis) over training time (x-axis).
```
